docs/src/main/sphinx/develop/table-functions.rst

```diff
@@ -0,0 +1,133 @@
+===============
+Table functions
+===============
+
+Table function declaration
+--------------------------
+
+Trino supports adding custom table functions. They are declared by connectors
+through implementing dedicated interfaces.
+
+To declare a table function, you need to subclass ``ConnectorTableFunction``.
+
+The constructor
+^^^^^^^^^^^^^^^
+
+.. code-block:: java
+
+    public class MyFunction
+            extends ConnectorTableFunction
+    {
+        public MyFunction()
+        {
+            super(
+                    "system",
+                    "my_function",
+                    List.of(
+                            new ScalarArgumentSpecification("column_count", INTEGER, 2),
+                            new ScalarArgumentSpecification("row_count", INTEGER)),
+                    GENERIC_TABLE);
+        }
+    }
+
+The constructor takes the following arguments:
+
+- schema name
```

Comment:
how do you decide what schema value to use? what impact does that have on the user?

Reply:
added explanation